### PR TITLE
Enhancement in Contact-us form done!

### PR DESCRIPTION
**Pull request by ShubhamSj07**
## Issue that this pull request solves #505 

 Closes: # (issue number)

### Brief description of what is fixed or changed
Added Validations to contact-form , git-icon hover effect updated also updated validation on email for newsletter section.
I had used functional component based approach

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
ContactForm

## Other information

Any other information that is important to this pull request

### CODE REVIEW

**Comment by Kajol-Kumari** [comment]
Hey @ShubhamSj07 please resolve the conflict here so that I can test and review it.

**Comment by ShubhamSj07** [comment]
@Kajol-Kumari , conflict resolved pls review it now..

File: frontend/src/components/Footer/Footer.jsx [hunk 1]
```diff
@@ -1,255 +1,263 @@
...
-        )
-      ) {
-        formIsValid = false;
-        errors["email"] = "* Email is not valid";
-      }
+    let isValid = true;
+
+    //email
+    // eslint-disable-next-line
+    const emailRegex = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (Kajol-Kumari) [inline frontend/src/components/Footer/Footer.jsx:20]: Instead of using regex, let's use `joi-validators` only in order to be consistent everywhere :)

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @ShubhamSj07 The validation is working but it needs Enhamcements - 
1. The validation should be trigerred `onCHange` but until we click on `submit` button the check isn't being done here.
2. The error message should be shown near to the respective input fields. (check for the bottom-margin of input tag) 
3. The card that we are showing after submission needs improvement. Please take the reference of the card below for its UI.
Screenshot from 2021-03-16 11-11-23

**Comment by ShubhamSj07** [comment]
@Kajol-Kumari , yes we should be having onChange validations , i will try to fix this at earliest.
would u pls describe more on ui of card after submission we gonna show, like i had included text provided : We Have Heard You!!
We Will Soon Get Back To You If Required!

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari , yes we should be having onChange validations , i will try to fix this at earliest.
> would u pls describe more on ui of card after submission we gonna show, like i had included text provided : We Have Heard You!! We Will Soon Get Back To You If Required!
> should i include any links for same ?

Hey! you should work on improving it's UI. like make the card appear in the middle of the page, the heading of the card should be `Hello There!` and keep the content as `We have heard you!(maybe put some emoji here) We will get back to you very soon if required!` Take the reference of the card I have attach above for the reference of the UI. 

P.S - We don't need button in the card here.

File: frontend/src/components/Footer/Footer.jsx (hunk 1 above)
Comment (ShubhamSj07) [reply frontend/src/components/Footer/Footer.jsx:20]: @Kajol-Kumari , earlier in footer section email was validated by regex  in class component, which I had updated in functional component.I hadn't use joi-validator dependency earlier!
Have we use joi in this project ,so I could take ref.

**Comment by ShubhamSj07** [comment]
@Kajol-Kumari , i had added **onChange** validations on input fields , also please review design of card
Pls guide me for further steps.
HITK_goodbye_card

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari , i had added **onChange** validations on input fields , also please review design of card
> Pls guide me for further steps.
> HITK_goodbye_card

Hey @ShubhamSj07 The UI looks good to me. Feel free to push the changes

File: frontend/src/components/Footer/Footer.jsx (hunk 1 above)
Comment (Kajol-Kumari) [reply frontend/src/components/Footer/Footer.jsx:20]: Hey @ShubhamSj07 please take the reference of resource form for using `joi-validator` and in order to be consistent we are going with it.

**Comment by ShubhamSj07** [comment]
@Kajol-Kumari , email validations by joi-validator is not working properly as we type for eg: demo@s still it is considered as valid email ,also if i use regex it shows in err msg like email should match whole regex expression!
So, should i create seperate validations for email as i had done before it was working perfectly 
Else, everything is working fine!

**Comment by ShubhamSj07** [comment]
@Kajol-Kumari , Facing error

Err

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari , Facing error Please help!
> 
> Err

Hey @Shubham-Kumar-2000 You need to pull the changes first using [this](https://stackoverflow.com/questions/53712930/how-to-fix-this-in-git-this-branch-is-out-of-date-with-the-base-branch) and then you need to resolve the conflicts manually by navigating to each conflicting files and then commit and push the changes

**Comment by Shubham-Kumar-2000** [comment]
@Kajol-Kumari  why should I resolve the conflict. 

I think you have tagged me by mistake.

**Comment by ShubhamSj07** [comment]
> @Kajol-Kumari why should I resolve the conflict.
> 
> I think you have tagged me by mistake.

Yes, by mistakenly she had tagged you
Nothing much to worry.

**Comment by Kajol-Kumari** [comment]
> @Kajol-Kumari why should I resolve the conflict.
> 
> I think you have tagged me by mistake.

Sorry for that :)

**Comment by ShubhamSj07** [comment]
Hey @Kajol-Kumari , I am not able to push my changes also how did this issue has been closed?
I am ready with all the changes but facing errors while pushing code

**Comment by jackfrost13** [comment]
Hey, I see you have closed this PR. I cannot open the pr.
Please push your changes to your branch, probably then might get the option to reopen the pr

**Pull request closed without merging**